core/src/parking_lot.rs

```diff
@@ -104,14 +121,21 @@ impl FairTimeout {
     // Determine whether we should force a fair unlock, and update the timeout
     #[inline]
     fn should_timeout(&mut self) -> bool {
-        let now = Instant::now();
-        if now > self.timeout {
-            // Time between 0 and 1ms.
-            let nanos = self.gen_u32() % 1_000_000;
-            self.timeout = now + Duration::new(0, nanos);
-            true
-        } else {
-            false
+        #[cfg(not(all(
+            target_arch = "wasm32",
```

Comment:
```suggestion
            target_family = "wasm",
```

Reply:
Though this is only for up to date compilers, but it's future proof towards wasm64.